src/index.js

```diff
@@ -138,12 +138,11 @@ module.exports = class Bitwap {
       cb(err, block)
     }
 
-    this.getBlocks([key], (err, res) => {
-      if (err) {
-        return done(err)
-      }
+    this.getBlocks([key], (results) => {
+      const err = results[key].error
+      const block = results[key].block
 
-      done(null, res[0])
+      done(err, block)
```

Comment:
No witchcraft, what do you not understand? If there is no `err` it 'll be passed as `done(undefined, block)` and otherwise the `err` will be the first arg